specification/cognitiveservices/Language.Conversations/common.tsp

```diff
@@ -0,0 +1,292 @@
+@service({
+    title: "Microsoft Cognitive Language Service",
+    version: "2022-05-01",
+})
+namespace Azure.AI.Language.Conversations;
+
+@doc("Error Response")
+model ErrorResponse {
+    @doc("The error object.")
+    error: Error;
+}
+
+@doc("The error object.")
+model Error extends Record<unknown> {
+    @doc("One of a server-defined set of error codes.")
+    code: ErrorCode;
+
+    @doc("A human-readable representation of the error.")
+    message: string;
+
+    @doc("The target of the error.")
+    target?: string;
+
+    @doc("An array of details about specific errors that led to this reported error.")
+    details?: Array<Error>;
+
+    @doc("An object containing more specific information than the current object about the error.")
+    innererror?: InnerErrorModel;
+}
+
+@doc("An object containing more specific information about the error. As per Microsoft One API guidelines - https://github.com/Microsoft/api-guidelines/blob/vNext/Guidelines.md#7102-error-condition-responses.")
+model InnerErrorModel extends Record<unknown> {
+    @doc("One of a server-defined set of error codes.")
+    code: InnerErrorCode;
+
+    @doc("Error message.")
+    message: string;
+
+    @doc("Error details.")
+    details?: Record<string>;
+
+    @doc("Error target.")
+    target?: string;
+
+    @doc("An object containing more specific information than the current object about the error.")
+    innererror?: InnerErrorModel;
+}
+
+@doc("Language of the text records. This is BCP-47 representation of a language. For example, use \"en\" for English; \"es\" for Spanish etc. If not set, use \"en\" for English as default.")
+model Language {}
+
+@doc("Specifies the method used to interpret string offsets.  Defaults to Text Elements (Graphemes) according to Unicode v8.0.0. For additional information see https://aka.ms/text-analytics-offsets.")
+enum StringIndexType {
+    @doc("Returned offset and length values will correspond to TextElements (Graphemes and Grapheme clusters) confirming to the Unicode 8.0.0 standard. Use this option if your application is written in .Net Framework or .Net Core and you will be using StringInfo.")
+    TextElements_v8,
+
+    @doc("Returned offset and length values will correspond to Unicode code points. Use this option if your application is written in a language that support Unicode, for example Python.")
+    UnicodeCodePoint,
+
+    @doc("Returned offset and length values will correspond to UTF-16 code units. Use this option if your application is written in a language that support Unicode, for example Java, JavaScript.")
+    Utf16CodeUnit,
+}
+
+@doc("Represents List of Question Answers.")
+model AnswersResult {
+    @doc("Represents Answer Result list.")
+    answers: Array<KnowledgeBaseAnswer>;
```

Comment:
For this an all properties, unless a model in swagger has a property in `required: []`, it should be optional e.g.,
```suggestion
    answers?: Array<KnowledgeBaseAnswer>;
```
As with Cognitive now, almost every property should be this way.